### DISCUSSION TITLE
Linux distribution-independent install script

### DISCUSSION THREAD

**Discussion by rumbleFTW** [Emory-HITI/Niffler#374] [Ideas] [answered]
I wanted the Niffler install script to be distribution independent. The current installer uses yum, which is the package manager for Red-hat linux distros. Is there any future plan to do so?

**Comment by pradeeban** [comment, marked as answer]
Thanks for #375 towards fixing this. We did not have sufficient test environments to make and test to make it distribution-independent.

**Comment by rumbleFTW** [comment]
Thanks for merging the PR. I authored the commit, but it still doesn't support some lesser-known distributions. Would love to see someone pick-up the work where I left off and find an elegant solution to this.

**Reply by pradeeban** [reply]
We are a research university project. Your PR already handles the popular Linux distros. That is good enough for me. We don't need to handle all the Linux distros. I would like to see Windows and MacOS versions more, though.

**Reply by yuvraj-wale** [reply]
Hi @pradeeban, I was considering creating a separate installation script for Niffler on macOS, perhaps named install_macos.sh or something similar. I wanted to confirm if this would be the appropriate way to proceed.

**Reply by pradeeban** [reply]
@yuvraj-wale, that is perfect. Go ahead!

**Reply by yuvraj-wale** [reply]
hey @pradeeban while I was working on the install script for macOS I came across something called 'disable thp' which is related to memory allocation in linux kernel, from what I know macOS doesn't need this. Am I right?

**Reply by pradeeban** [reply]
yes, you can ignore that for Mac.

**Reply by yuvraj-wale** [reply]
@pradeeban we will have to create a mdextractor.plist file under meta-extraction/service in order for these install.sh commands to execute 
```
    sudo cp modules/meta-extraction/service/mdextractor.service /etc/systemd/system/
    sudo systemctl daemon-reload
    sudo systemctl start mdextractor.service
    sudo systemctl enable mdextractor.service
```
that to under `/Library/LaunchDaemons/` instead of `/etc/systemd/system/` in the first line in order to execute for macOS. Am I thinking right? , if so could you guide me little on how should I proceed further maybe how to make a .plist file which would be macOS equivalent of mdextractor.service file.

**Reply by yuvraj-wale** [reply]
by "for these install.sh command to execute", I mean the macOS equivalent of install.sh command to execute ie:- macOS_install.sh

**Reply by pradeeban** [reply]
I have not done MacOS installation scripts. But I was thinking something along the lines of https://medium.com/swlh/the-easiest-way-to-build-macos-installer-for-your-application-34a11dd08744

A script that can create an installable package for MacOS, which we can even release from the binaries here (once we test it sufficiently).

**Reply by yuvraj-wale** [reply]
ok cool, I will check this out.

**Comment by H9660** [comment]
Hi @pradeeban can I start making the proposal for this project for you to review them?

**Reply by pradeeban** [reply]
Sure. The project ideas are listed at https://github.com/NISYSLAB/Emory-BMI-GSoC/blob/main/README.md

**Reply by H9660** [reply]
Thanks
